sdk/servicebus/Azure.Messaging.ServiceBus/tests/Diagnostics/DiagnosticScopeLiveTests.cs

```diff
@@ -0,0 +1,439 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.Tests;
+using Azure.Messaging.ServiceBus.Diagnostics;
+using NUnit.Framework;
+
+namespace Azure.Messaging.ServiceBus.Tests.Diagnostics
+{
+    [NonParallelizable]
+    public class DiagnosticScopeLiveTests : ServiceBusLiveTestBase
+    {
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task SenderReceiverActivities(bool useSessions)
```

Comment:
Wow.   This is a very busy test that covers quite a few different operations.   Is there a reason why you chose to use a single test rather than having a test dedicated to each individual operation?   I believe the latter would be easier to read, reason about, and maintain going forward.

Reply:
The main reason is that since the tests can't be run in parallel, I wanted to avoid repeating common setup steps like establishing a connection, sending messages, that would need to be repeated in each test.